src/sql/workbench/parts/notebook/models/notebookModel.ts

```diff
@@ -292,7 +297,24 @@ export class NotebookModel extends Disposable implements INotebookModel {
 				this._defaultLanguageInfo = this.getDefaultLanguageInfo(contents);
 				this._savedKernelInfo = this.getSavedKernelInfo(contents);
 				if (contents.cells && contents.cells.length > 0) {
-					this._cells = contents.cells.map(c => factory.createCell(c, { notebook: this, isTrusted: isTrusted }));
+					this._cells = contents.cells.map(c => {
+						let cellContent = (<nb.ICellContents>c);
+						if (c && cellContent && cellContent.cell_type === CellTypes.Markdown) {
+							this._textCellsLoading++;
+						}
+						let cellModel = factory.createCell(c, { notebook: this, isTrusted: isTrusted });
+						this._register(cellModel.onLoaded((e) => {
+							this._textCellsLoading--;
+							if (this._textCellsLoading <= 0) {
+								let editorOpenedTimestamp = this.telemetryService.getTimestampForEvent('editorOpened');
```

Comment:
How do we know there wasn't a different editor opened in the meantime? How does this work, does it get the last event with this ID?

Reply:
Actually - I see what you did in the below code. I don't think we should edit the core telemetry code to handle this. How about the following:
- In the NotebookInput constructor, create a timestamp to track opening and have a public `get` method for it
- In notebook.component.ts, pass this in as a notebook model option. Keep this as an optional field
- Use this in the model load code.

That way we avoid deep infra changes just to track "I requested open the editor, to all parts are loaded". Sound OK?